lib/Bio/DB/HTS.pm

```diff
@@ -2143,20 +2143,24 @@ sub index {
     return $self->index_open_in_safewd($fh) if Bio::DB::HTSfile->is_remote($path);
 
     if ($autoindex) {
-        if ( !( -e "${path}.bai" or -e "${path}.crai" ) ) {
+        if ( !( -e "${path}.bai" or -e "${path}.csi" or -e "${path}.crai" ) ) {
             $self->reindex($path);
         }
         elsif ( -e "${path}.bai" && mtime($path) > mtime("${path}.bai") ) {
             $self->reindex($path);
         }
+        elsif ( -e "${path}.csi" && mtime($path) > mtime("${path}.csi") ) {
+            croak "csi index is older than bam file, Bio::DB::HTS cannot index csi format."
+        }
         elsif ( -e "${path}.crai" && mtime($path) > mtime("${path}.crai") ) {
             $self->reindex($path);
         }
     }
 
     croak "No index file for $path; try opening file with -autoindex"
       unless -e "${path}.bai" or
-      -e "${path}.crai";
+        -e "${path}.csi" or
+        -e "${path}.crai";
```

Comment:
Aren't you actually suggesting to fail in the csi case by telling to rerun with autoindex?

Reply:
I'm assuming at some point `-autoindex` will allow csi.

Reply:
Fair enough.